### PR TITLE
ticdc: add description for cdc behaviour change

### PR DESCRIPTION
**Pull request by lidezhu**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by flowbehappy — APPROVED** [review]

File: ticdc/ticdc-behavior-change.md [hunk 1]
```diff
@@ -52,3 +52,44 @@ COMMIT;
 在上述示例中，通过执行三条 SQL 语句对两行数据的主键进行交换，但 TiCDC 只会接收到两条 Update 变更事件，即将主键 `a` 从 `1` 变更为 `2`，将主键 `a` 从 `2` 变更为 `1`，如果 MYSQL Sink 直接将这两条 Update 事件写入下游，会出现主键冲突的问题，导致 changefeed 报错。
 
 因此，TiCDC 会将这两条事件拆分为四条事件，即删除记录 `(1, 1)` 和 `(2, 2)` 以及写入记录 `(2, 1)` 和 `(1, 2)`。
+
+从 v8.1.0 开始，使用 MySQL Sink 时，TiCDC 在启动时会从 PD 获取一个当前的时间戳（记为 `thresholdTs`），对于 `commitTS` 小于 `thresholdTs` 的 Update 事件，在写入 Sorter 之前会将其拆分为 Delete 和 Insert 两条事件，这样可以确保同一事务内所有事件按照 Delete 事件在 Insert 事件之前的顺序进行排序。对于 `commitTS` 大于等于 `thresholdTs` 的 Update 事件，TiCDC 不会对其进行拆分。详情见 GitHub issue [#10918](https://github.com/pingcap/tiflow/issues/10918)。
```
Comment (qiancai) [inline ticdc/ticdc-behavior-change.md:56]:
```suggestion
从 v8.1.0 开始，使用 MySQL Sink 时，TiCDC 在启动时会从 PD 获取一个当前的时间戳 `thresholdTs`：

- 对于一个含有多条变更的事务，如果 Update 事件的主键或者非空唯一索引的列值发生改变且 `commitTS` 小于 `thresholdTs`，在写入 Sorter 模块之前 TiCDC 会将每条 Update 事件 拆分为 Delete 和 Insert 两条事件，以确保同一事务内所有事件按照 Delete 事件在 Insert 事件之前的顺序进行排序。
- 对于 `commitTS` 大于或等于 `thresholdTs` 的 Update 事件，TiCDC 不会对其进行拆分。详情见 GitHub issue [#10918](https://github.com/pingcap/tiflow/issues/10918)。
```

File: ticdc/ticdc-behavior-change.md [hunk 2]
```diff
@@ -52,3 +52,44 @@ COMMIT;
...
+
+该变更是由于 TiCDC 目前无法获取上游同一事务内多个 Update 事件之间的执行顺序，对于一个含有多条 Update 事件的事务，如果 Update 事件的主键或者非空唯一索引的列值发生改变，将 Update 事件拆分为 Delete 和 Insert 两条事件发往下游，可能会造成数据不一致问题。
```
Comment (qiancai) [inline ticdc/ticdc-behavior-change.md:58]:
```suggestion
该变更是由于 TiCDC 目前无法获取上游同一事务内多个 Update 事件之间的执行顺序，如果获取的 Update 事件顺序不正确，将 Update 事件拆分为 Delete 和 Insert 两条事件发往下游可能会造成数据不一致问题。
```

File: ticdc/ticdc-behavior-change.md [hunk 3]
```diff
@@ -52,3 +52,44 @@ COMMIT;
...
+INSERT INTO t VALUES (1, 1);
+INSERT INTO t VALUES (2, 2);
+
+BEGIN;
+UPDATE t SET a = 3 WHERE a = 2;
+UPDATE t SET a = 2 WHERE a = 1;
+COMMIT;
+```
+
+在上述示例中，事务内的两条 SQL 的执行顺序有先后依赖关系，即将主键 `a` 从 `2` 变更为 `3`，再将主键 `a` 从 `1` 变更为 `2`，如果 TiCDC 内部收到的 Update 事件顺序与事务内部实际的执行顺序不同，此时把它们分别拆分为 Delete 和 Insert 事件发往下游，会造成数据不一致问题。
```
Comment (qiancai) [inline ticdc/ticdc-behavior-change.md:73]:
```suggestion
在上述示例中，事务内的两条 SQL 的执行顺序有先后依赖关系，即先将主键 `a` 从 `2` 变更为 `3`，再将主键 `a` 从 `1` 变更为 `2`。如果 TiCDC 内部收到的 Update 事件顺序与上游事务内部实际的执行顺序不同，把收到的 Update 事件分别拆分为 Delete 和 Insert 事件发往下游，会造成数据不一致问题。
```

**Review by qiancai — APPROVED** [review]

File: ticdc/ticdc-behavior-change.md [hunk 4]
```diff
@@ -52,3 +52,66 @@ COMMIT;
...
+INSERT INTO t VALUES (1, 1);
+INSERT INTO t VALUES (2, 2);
+
+BEGIN;
+UPDATE t SET a = 3 WHERE a = 2;
+UPDATE t SET a = 2 WHERE a = 1;
+COMMIT;
+```
+
+在该示例中，事务内的两条 SQL 的执行顺序有先后依赖关系，即先将主键 `a` 从 `2` 变更为 `3`，再将主键 `a` 从 `1` 变更为 `2`。执行完该事务后，上游数据库内的记录为 `(2, 1)` 和 `(3, 2)`。
```
Comment (qiancai) [inline ticdc/ticdc-behavior-change.md:78]:
```suggestion
在该示例中，事务内的两条 Update 语句的执行顺序有先后依赖关系，即先将主键 `a` 从 `2` 变更为 `3`，再将主键 `a` 从 `1` 变更为 `2`。执行完该事务后，上游数据库内的记录为 `(2, 1)` 和 `(3, 2)`。
```

File: ticdc/ticdc-behavior-change.md [hunk 5]
```diff
@@ -49,6 +49,69 @@ UPDATE t SET a = 2 WHERE a = 3;
...
-在上述示例中，通过执行三条 SQL 语句对两行数据的主键进行交换，但 TiCDC 只会接收到两条 Update 变更事件，即将主键 `a` 从 `1` 变更为 `2`，将主键 `a` 从 `2` 变更为 `1`，如果 MYSQL Sink 直接将这两条 Update 事件写入下游，会出现主键冲突的问题，导致 changefeed 报错。
+在上述示例中，通过执行三条 SQL 语句对两行数据的主键进行交换，但 TiCDC 只会接收到两条 `UPDATE` 变更事件，即将主键 `a` 从 `1` 变更为 `2`，将主键 `a` 从 `2` 变更为 `1`，如果 MYSQL Sink 直接将这两条 `UPDATE` 事件写入下游，会出现主键冲突的问题，导致 changefeed 报错。
 
 因此，TiCDC 会将这两条事件拆分为四条事件，即删除记录 `(1, 1)` 和 `(2, 2)` 以及写入记录 `(2, 1)` 和 `(1, 2)`。
+
+#### MySQL Sink
+
+从 v8.1.0 开始，当使用 MySQL Sink 时，TiCDC 在启动时会从 PD 获取一个当前的时间戳 `thresholdTs`，并根据时间戳的值决定是否拆分 `UPDATE` 事件：
+
+- 对于一个含有多条变更的事务，如果 `UPDATE` 事件的主键或者非空唯一索引的列值发生改变，且事务的 `commitTS` 小于 `thresholdTs`，在写入 Sorter 模块之前 TiCDC 会将每条 `UPDATE` 事件拆分为 `DELETE` 和 `INSERT` 两条事件。
```
Comment (lidezhu) [inline ticdc/ticdc-behavior-change.md:60]:
```suggestion
- 如果 `UPDATE` 事件的主键或者非空唯一索引的列值发生改变，且事务的 `commitTS` 小于 `thresholdTs`，在写入 Sorter 模块之前 TiCDC 会将每条 `UPDATE` 事件拆分为 `DELETE` 和 `INSERT` 两条事件。
```

File: ticdc/ticdc-behavior-change.md [hunk 6]
```diff
@@ -49,6 +49,69 @@ UPDATE t SET a = 2 WHERE a = 3;
 COMMIT;
 ```
 
-在上述示例中，通过执行三条 SQL 语句对两行数据的主键进行交换，但 TiCDC 只会接收到两条 Update 变更事件，即将主键 `a` 从 `1` 变更为 `2`，将主键 `a` 从 `2` 变更为 `1`，如果 MYSQL Sink 直接将这两条 Update 事件写入下游，会出现主键冲突的问题，导致 changefeed 报错。
+在上述示例中，通过执行三条 SQL 语句对两行数据的主键进行交换，但 TiCDC 只会接收到两条 `UPDATE` 变更事件，即将主键 `a` 从 `1` 变更为 `2`，将主键 `a` 从 `2` 变更为 `1`，如果 MYSQL Sink 直接将这两条 `UPDATE` 事件写入下游，会出现主键冲突的问题，导致 changefeed 报错。
 
 因此，TiCDC 会将这两条事件拆分为四条事件，即删除记录 `(1, 1)` 和 `(2, 2)` 以及写入记录 `(2, 1)` 和 `(1, 2)`。
+
+#### MySQL Sink
```
Comment (lidezhu) [inline ticdc/ticdc-behavior-change.md:56]:
```suggestion
### MySQL Sink
```

File: releases/release-8.1.0.md [hunk 7]
```diff
@@ -172,6 +172,7 @@ TiDB 8.1.0 为长期支持版本 (Long-Term Support Release, LTS)。
 
 * 在之前的版本中，TiDB Lightning 的配置项 `tidb.tls` 在取值为 `"false"` 和 `""` 时的行为是相同的，在取值为 `"skip-verify"` 和 `"preferred"` 时的行为也是相同的。从 v8.1.0 开始，TiDB Lightning 对 `tidb.tls` 取值为 `"false"`、`""`、`"skip-verify"` 和 `"preferred"` 时的行为进行了区分。更多信息，请参考 [TiDB Lightning 配置参数](/tidb-lightning/tidb-lightning-configuration.md)。
 * 对于设置了 `AUTO_ID_CACHE=1` 的表，TiDB 支持[中心化分配自增 ID 服务](/auto-increment.md#mysql-兼容模式)。在之前的版本中，该服务的“主”TiDB 节点在进程退出（如该 TiDB 节点重启）时会自动执行 `forceRebase` 操作，以确保自动分配的 ID 尽可能连续。然而，当设置过 `AUTO_ID_CACHE=1` 的表过多时，执行 `forceRebase` 会非常耗时，导致 TiDB 无法及时重启，甚至阻塞数据写入，影响系统可用性。因此，从 v8.1.0 起，TiDB 取消了 `forceRebase` 操作，解决了上述问题，但会造成主备切换期间部分自动分配的 ID 出现不连续。
+* 在之前的版本中，TiCDC 在处理包含 `UPDATE` 事件的事务时，如果事件的主键或者非空唯一索引的列值发生改变，则会将该条事件拆分为 `DELETE` 和 `INSERT` 两条事件。从 v8.1.0 开始，当使用 MySQL Sink 时，如果 `UPDATE` 事件的主键或者非空唯一索引的列值发生改变，且事务的 `commitTS` 小于 TiCDC 启动时的时间戳 `thresholdTs`，TiCDC 才会将该 `UPDATE` 事件拆分为 `DELETE` 和 `INSERT` 两条事件，然后写入 Sorter 模块。该行为变更解决了由于 TiCDC 接收到的 `UPDATE` 事件顺序可能不正确，导致拆分后的 `DELETE` 和 `INSERT` 事件顺序也可能不正确，从而引发下游数据不一致的问题。更多信息，请参考[用户文档](/ticdc/ticdc-behavior-change.md#mysql-sink)。
```
Comment (lidezhu) [inline releases/release-8.1.0.md:175]:
```suggestion
* 在之前的版本中，TiCDC 在处理包含 `UPDATE` 事件的事务时，如果事件的主键或者非空唯一索引的列值发生改变，则会将该条事件拆分为 `DELETE` 和 `INSERT` 两条事件。从 v8.1.0 开始，当使用 MySQL Sink 时，如果 `UPDATE` 事件的主键或者非空唯一索引的列值发生改变，且事务的 `commitTS` 小于 TiCDC 启动时从 PD 获取的当前时间戳 `thresholdTs`，TiCDC 才会将该 `UPDATE` 事件拆分为 `DELETE` 和 `INSERT` 两条事件，然后写入 Sorter 模块。该行为变更解决了由于 TiCDC 接收到的 `UPDATE` 事件顺序可能不正确，导致拆分后的 `DELETE` 和 `INSERT` 事件顺序也可能不正确，从而引发下游数据不一致的问题。更多信息，请参考[用户文档](/ticdc/ticdc-behavior-change.md#mysql-sink)。
```

File: ticdc/ticdc-behavior-change.md [hunk 8]
```diff
@@ -49,6 +49,69 @@ UPDATE t SET a = 2 WHERE a = 3;
...
-在上述示例中，通过执行三条 SQL 语句对两行数据的主键进行交换，但 TiCDC 只会接收到两条 Update 变更事件，即将主键 `a` 从 `1` 变更为 `2`，将主键 `a` 从 `2` 变更为 `1`，如果 MYSQL Sink 直接将这两条 Update 事件写入下游，会出现主键冲突的问题，导致 changefeed 报错。
+在上述示例中，通过执行三条 SQL 语句对两行数据的主键进行交换，但 TiCDC 只会接收到两条 `UPDATE` 变更事件，即将主键 `a` 从 `1` 变更为 `2`，将主键 `a` 从 `2` 变更为 `1`，如果 MYSQL Sink 直接将这两条 `UPDATE` 事件写入下游，会出现主键冲突的问题，导致 changefeed 报错。
 
 因此，TiCDC 会将这两条事件拆分为四条事件，即删除记录 `(1, 1)` 和 `(2, 2)` 以及写入记录 `(2, 1)` 和 `(1, 2)`。
+
+### MySQL Sink
+
+从 v8.1.0 开始，当使用 MySQL Sink 时，TiCDC 在启动时会从 PD 获取一个当前的时间戳 `thresholdTs`，并根据时间戳的值决定是否拆分 `UPDATE` 事件：
+
+- 如果 `UPDATE` 事件的主键或者非空唯一索引的列值发生改变，且事务的 `commitTS` 小于 `thresholdTs`，在写入 Sorter 模块之前 TiCDC 会将每条 `UPDATE` 事件拆分为 `DELETE` 和 `INSERT` 两条事件。
```
Comment (qiancai) [inline ticdc/ticdc-behavior-change.md:60]:
```suggestion
- 对于含有单条或多条 `UPDATE` 变更的事务，如果 `UPDATE` 事件的主键或者非空唯一索引的列值发生改变，且事务的 `commitTS` 小于 `thresholdTs`，在写入 Sorter 模块之前 TiCDC 会将每条 `UPDATE` 事件拆分为 `DELETE` 和 `INSERT` 两条事件。
```

File: releases/release-8.1.0.md [hunk 9]
```diff
@@ -172,6 +172,7 @@ TiDB 8.1.0 为长期支持版本 (Long-Term Support Release, LTS)。
 
 * 在之前的版本中，TiDB Lightning 的配置项 `tidb.tls` 在取值为 `"false"` 和 `""` 时的行为是相同的，在取值为 `"skip-verify"` 和 `"preferred"` 时的行为也是相同的。从 v8.1.0 开始，TiDB Lightning 对 `tidb.tls` 取值为 `"false"`、`""`、`"skip-verify"` 和 `"preferred"` 时的行为进行了区分。更多信息，请参考 [TiDB Lightning 配置参数](/tidb-lightning/tidb-lightning-configuration.md)。
 * 对于设置了 `AUTO_ID_CACHE=1` 的表，TiDB 支持[中心化分配自增 ID 服务](/auto-increment.md#mysql-兼容模式)。在之前的版本中，该服务的“主”TiDB 节点在进程退出（如该 TiDB 节点重启）时会自动执行 `forceRebase` 操作，以确保自动分配的 ID 尽可能连续。然而，当设置过 `AUTO_ID_CACHE=1` 的表过多时，执行 `forceRebase` 会非常耗时，导致 TiDB 无法及时重启，甚至阻塞数据写入，影响系统可用性。因此，从 v8.1.0 起，TiDB 取消了 `forceRebase` 操作，解决了上述问题，但会造成主备切换期间部分自动分配的 ID 出现不连续。
+* 在之前的版本中，TiCDC 在处理包含 `UPDATE` 事件的事务时，如果事件的主键或者非空唯一索引的列值发生改变，则会将该条事件拆分为 `DELETE` 和 `INSERT` 两条事件。从 v8.1.0 开始，当使用 MySQL Sink 时，如果 `UPDATE` 事件的主键或者非空唯一索引的列值发生改变，且事务的 `commitTS` 小于 TiCDC 启动时从 PD 获取的当前时间戳 `thresholdTs`，TiCDC 才会将该 `UPDATE` 事件拆分为 `DELETE` 和 `INSERT` 两条事件，然后写入 Sorter 模块。该行为变更解决了由于 TiCDC 接收到的 `UPDATE` 事件顺序可能不正确，导致拆分后的 `DELETE` 和 `INSERT` 事件顺序也可能不正确，从而引发下游数据不一致的问题。更多信息，请参考[用户文档](/ticdc/ticdc-behavior-change.md#mysql-sink)。
```
Comment (qiancai) [inline releases/release-8.1.0.md:175]:
```suggestion
* 在之前的版本中，TiCDC 在处理包含 `UPDATE` 变更的事务时，如果 `UPDATE` 事件的主键或者非空唯一索引的列值发生改变，则会将该条事件拆分为 `DELETE` 和 `INSERT` 两条事件。从 v8.1.0 开始，当使用 MySQL Sink 时，如果 `UPDATE` 事件的主键或者非空唯一索引的列值发生改变，且事务的 `commitTS` 小于 TiCDC 启动时从 PD 获取的当前时间戳 `thresholdTs`，TiCDC 才会将该 `UPDATE` 事件拆分为 `DELETE` 和 `INSERT` 两条事件，然后写入 Sorter 模块。该行为变更解决了由于 TiCDC 接收到的 `UPDATE` 事件顺序可能不正确，导致拆分后的 `DELETE` 和 `INSERT` 事件顺序也可能不正确，从而引发下游数据不一致的问题。更多信息，请参考[用户文档](/ticdc/ticdc-behavior-change.md#mysql-sink)。
```

File: ticdc/ticdc-behavior-change.md [hunk 10]
```diff
@@ -49,6 +49,69 @@ UPDATE t SET a = 2 WHERE a = 3;
...
+    COMMIT;
+    ```
+
+    下游执行完该事务后，下游数据库内的记录和上游数据库一样，都为 `(2, 1)` 和 `(3, 2)`，保证了数据一致性。
+
+从该示例中可以看到，在写入 Sorter 模块之前将 `UPDATE` 事件拆分为 `DELETE` 和 `INSERT` 事件，可以保证拆分后所有的 `DELETE` 事件都在 `INSERT` 事件之前执行，这样无论 TiCDC 收到的 `UPDATE` 事件顺序，均可以保证数据一致性。
+
+> **注意：**
+>
+> 该行为变更后，在使用 MySQL Sink 时，TiCDC 在大部分情况下都不会拆分 `UPDATE` 事件，因此 changefeed 在运行时可能会出现主键或唯一键冲突的问题。该问题会导致 changefeed 自动重启，重启后发生冲突的 `UPDATE` 事件会被拆分为 `DELETE` 和 `INSERT` 事件并写入 Sorter 模块中，此时可以确保同一事务内所有事件按照 `DELETE` 事件在 `INSERT` 事件之前的顺序进行排序从而正确完成数据同步。
```
Comment (Oreoxmt) [inline ticdc/ticdc-behavior-change.md:117]:
```suggestion
> 该行为变更后，在使用 MySQL Sink 时，TiCDC 在大部分情况下都不会拆分 `UPDATE` 事件，因此 changefeed 在运行时可能会出现主键或唯一键冲突的问题。该问题会导致 changefeed 自动重启，重启后发生冲突的 `UPDATE` 事件会被拆分为 `DELETE` 和 `INSERT` 事件并写入 Sorter 模块中，此时可以确保同一事务内所有事件按照 `DELETE` 事件在 `INSERT` 事件之前的顺序进行排序，从而正确完成数据同步。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Pull request merged**